src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,50 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
 public class SalaryInfo {
+    private static DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static String WHITESPACE_DELIMITER = " ";
+    private Map<String, Integer> workerInfo = new LinkedHashMap<>();
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate localDateFrom = LocalDate.parse(dateFrom, formatter);
+        LocalDate localDateTo = LocalDate.parse(dateTo, formatter);
+        setWorkersName(workerInfo,names);
```

Comment:
You solution looks a little bit complicated, there is redundant to create Map for the logic. 

Let's consider this solution like an option

```suggestion
        StringBuilder result = new StringBuilder("Report for period ")
                .append(dateFrom).append(" - ").append(dateTo);
        for (String name : names) {
            int sum = 0;
            for (String datum : data) {
                String[] lineFromData = datum.split(DATA_LINE_SEPARATOR);
                LocalDate currentDate = LocalDate
                        .parse(lineFromData[INDEX_OF_DATE], dateTimeFormatter);
                if (name.equals(lineFromData[INDEX_OF_NAME]) && !currentDate.isAfter(localDateTo)
                        && !currentDate.isBefore(localDateFrom)) {
                    sum += Integer.parseInt(lineFromData[INDEX_OF_WORKING_HOURS])
                            * Integer.parseInt(lineFromData[INDEX_OF_INCOME_PER_HOUR]);
                }
            }
            result.append(System.lineSeparator()).append(name).append(" - ").append(sum);
        }
        return result.toString();
```